rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/AccessIdController.java

```diff
@@ -58,16 +66,20 @@ public AccessIdController(LdapClient ldapClient) {
 
   @GetMapping(path = Mapping.URL_ACCESSID_GROUPS)
   public ResponseEntity<List<AccessIdRepresentationModel>> getGroupsByAccessId(
-      @RequestParam("access-id") String accessId) throws InvalidArgumentException {
+      @RequestParam("access-id") String accessId)
+      throws InvalidArgumentException, NotAuthorizedException {
+
     LOGGER.debug("Entry to getGroupsByAccessId(access-id= {})", accessId);
+
+    taskanaEngine.checkRoleMembership(TaskanaRole.ADMIN, TaskanaRole.BUSINESS_ADMIN);
+
     if (!validateAccessId(accessId)) {
       throw new InvalidArgumentException("The accessId is invalid");
     }
-    List<AccessIdRepresentationModel> accessIdUsers;
+    List<AccessIdRepresentationModel> accessIds;
     ResponseEntity<List<AccessIdRepresentationModel>> response;
-    accessIdUsers = ldapClient.searchUsersAndGroups(accessId);
-    accessIdUsers.addAll(ldapClient.searchGroupsofUsersIsMember(accessId));
-    response = ResponseEntity.ok(accessIdUsers);
+    accessIds = ldapClient.searchGroupsAccessIdIsMemberOf(accessId);
+    response = ResponseEntity.ok(accessIds);
```

Comment:
Why declare the type and then assign the value? This can be optimized (you don't have to do this.. I just noticed it)